assets/eip-3525/test/test.ts

```diff
@@ -0,0 +1,374 @@
+  };
+
+  const mintWithOutDeploy = async (
+    erc3525: ERC3525BurnableUpgradeable,
+    minter: SignerWithAddress,
+    slot: string
+  ): Promise<TokenData> => {
+    const value = BigNumber.from("1000000000000000000");
+    await erc3525.mint(minter.address, slot, value);
+
+    let eventFilter = erc3525.filters["TransferValue"](0);
+    let block = await ethers.provider.getBlock("latest");
+    let event = await erc3525.queryFilter(eventFilter, block.number, "latest");
+    let args = event[0]["args"];
+    const tokenData = {
+      id: BigNumber.from(args[1]),
+      slot: BigNumber.from(slot),
+      balance: value,
+      owner: minter.address,
+      erc3525: erc3525,
+    };
+
+    return tokenData;
+  };
+
+  const checkTransferEvent = async (
+    erc3525: ERC3525BurnableUpgradeable,
+    from: string,
+    to: string,
+    tokenId: BigNumber
+  ) => {
+    let eventFilter = erc3525.filters["Transfer"](from, to);
+    let block = await ethers.provider.getBlock("latest");
+    let event = await erc3525.queryFilter(eventFilter, block.number, "latest");
+
+    let args = event[0]["args"];
+    expect(args[0]).to.equal(from);
+    expect(args[1]).to.equal(to);
+    expect(args[2]).to.equal(tokenId);
+  };
+
+  const checkTransferValueEvent = async (
+    erc3525: ERC3525BurnableUpgradeable,
+    fromTokenId: BigNumber,
+    toTokenId: BigNumber,
+    balance: BigNumber
+  ) => {
+    let eventFilter = erc3525.filters["TransferValue"](fromTokenId, toTokenId);
+    let block = await ethers.provider.getBlock("latest");
+    let event = await erc3525.queryFilter(eventFilter, block.number, "latest");
+    let args = event[0]["args"];
+    expect(args[0]).to.equal(fromTokenId);
+    expect(args[1]).to.equal(toTokenId);
+    expect(args[2]).to.equal(balance);
+  };
+
+  describe("ERC721 compatible interface", function () {
+    it("mint should be success", async function () {
+      const t = await mint();
+      await checkTransferEvent(t.erc3525, ZERO_ADDRESS, t.owner, t.id);
+
+      expect(await t.erc3525["balanceOf(address)"](t.owner)).to.eq(t.id);
+      expect(await t.erc3525.ownerOf(t.id)).to.eq(t.owner);
+      await expect(t.erc3525.ownerOf(5)).revertedWith(
+        "ERC3525: owner query for nonexistent token"
+      );
+      expect(await t.erc3525["balanceOf(uint256)"](t.id)).to.eq(t.balance);
+      expect(await t.erc3525.slotOf(t.id)).to.eq(t.slot);
+      expect(await t.erc3525.totalSupply()).to.eq(1);
+    });
+
+    it("approve all should be success", async () => {
+      const [_, approval] = await ethers.getSigners();
+
+      const t = await mint();
+
+      await t.erc3525.setApprovalForAll(approval.address, true);
+      expect(await t.erc3525.isApprovedForAll(t.owner, approval.address)).to.eq(
+        true
+      );
+      expect(
+        await t.erc3525.isApprovedForAll(
+          t.owner,
+          "0x000000000000000000000000000000000000dEaD"
+        )
+      ).to.eq(false);
+    });
```

Comment:
Ideally, the tests should be converted to a more portable format.  A good example of this is an array of test cases with expected outputs, exceptions, logs, and post-conditions for each.  At the moment, this test file comes with a lot of JS specific stuff and includes deployment code which is out of scope for a good set of test cases.  Note, the section is called "Test Cases" not "Test Infrastructure".  😀